reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/ZipWith.kt

```diff
@@ -0,0 +1,4 @@
+package com.badoo.reaktive.observable
+
+fun <T, R> Observable<T>.zipWith(observable: Observable<R>): Observable<Pair<T, R>> =
```

Comment:
It would be good to provide a zipper function, so the signature should be:
```
fun <T, S, R> Observable<T>.zipWith(other: Observable<S>, zipper: (T, S) -> R): Observable<R> = 
    zip(this, other, zipper)
```

And same for `Maybe` and `Single`.

Reply:
Also please rename `observable` to `other` to keep consistency with the reset of the code.

Reply:
Added the mappers for Observable, Single and Maybe. 
Also renamed the other stream to "other" in all of them.

Reply:
Added a variant with a mapper to the zipWith method.
Also renamed the other stream to "other" in all of them.